src/utils/index.ts

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import URL from 'url-parse';
+import Dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import minMax from 'dayjs/plugin/minMax';
+import { TDateTimeParser } from '../i18n/Streami18n';
+
+Dayjs.extend(utc);
+Dayjs.extend(minMax);
+
+// import type { UserResponse } from 'getstream';
+
+export function humanizeTimestamp(timestamp: string | Dayjs.Dayjs, tDateTimeParser: TDateTimeParser) {
+  let time;
+  // Following calculation is based on assumption that tDateTimeParser()
+  // either returns momentjs or dayjs object.
+
+  // When timestamp doesn't have z at the end, we are supposed to take it as UTC time.
+  // Ideally we need to adhere to RFC3339. Unfortunately this needs to be fixed on backend.
+  if (typeof timestamp === 'string' && timestamp[timestamp.length - 1].toLowerCase() === 'z') {
+    time = tDateTimeParser(timestamp);
+  } else {
+    time = tDateTimeParser(timestamp).add(Dayjs(timestamp).utcOffset(), 'minute'); // parse time as UTC
+  }
+
+  const now = tDateTimeParser();
+  return time.from(now);
+}
+
+type ErrorUser = { error: string };
+function isErrorUser(user: unknown | ErrorUser): user is ErrorUser {
+  return !!user && typeof (user as ErrorUser).error === 'string';
+}
+
+export function userOrDefault<T>(user?: T | string | { error: string } | null) {
```

Comment:
```suggestion
export function userOrDefault<T extends UserAPIResponse>(user?: T | string | { error: string } | null) {
```

Reply:
`UserAPIResponse` also needs a generic to work correctly, let's revisit this later if needed